Update template spelling and wording

dougbu opened the pull request:
- follow up to #46714
- align template.json and templatestrings.en.json files
- correct 'ReadWrite' and 'Am'
- build to create localized template files
  - good to do this whenever we create new templates

dougbu committed:
Update template spelling and wording
- follow up to #46714
- align template.json and templatestrings.en.json files
- correct 'ReadWrite' and 'Am'
- ask a few questions of reviewers

dougbu commented:
Noticed a few issues when glancing at #46882.

The changes contain a few comments for reviewers. Though I used loc comment syntax (start w/ an underscore and end w/ ".comment"), the comments are _not_ intended to be merged.

sayedihashimi commented:
It would be great to have @phenning to review the template related updates here.

phenning reviewed (commented):
From a Visual Studio perspective, LGTM. Note that we don't surface any of the item template metadata directly in VIsual Studio as of now anyway, as we still use the vstemplate -> template engine pattern rather than exposing the sdk template directly in the new item dialog.

sayedihashimi commented on src/ProjectTemplates/Web.ItemTemplates/content/MvcController/.template.config/template.json:
```diff
@@ -46,7 +46,7 @@
       "id": "openInEditor",
       "condition": "(HostIdentifier != \"dotnetcli\" && HostIdentifier != \"dotnetcli-preview\")",
-      "description": "Opens the created controller in the editor",
+      "description": "Opens HomeController.cs in the editor",
```
Same feedback regarding `HomeController.cs` as `ValueController` at https://github.com/dotnet/aspnetcore/pull/46886/files#r1119243746. When the user uses this template the generated file will be named by the user. `HomeController` is an implementation detail of the template and won't make sense to the end-user.

dougbu commented on src/ProjectTemplates/Web.ItemTemplates/content/ApiController/.template.config/template.json:
```diff
@@ -46,7 +46,7 @@
       "id": "openInEditor",
       "condition": "(HostIdentifier != \"dotnetcli\" && HostIdentifier != \"dotnetcli-preview\")",
-      "description": "Opens the created controller in the editor",
+      "description": "Opens ValueController.cs in the editor",
```
1. This name should be automatically updated to reflect the name of the file a user chooses.
2. Since this is an item template, the description should be about that item.

dougbu replied:
@phenning do you mean my point (1) is incorrect❔ If so, ASP.NET has a larger problem because most of our item templates use the specific filename from `primaryOutputs`.

phenning replied:
Yea, it will work in a primary output. I'm talking specifically about the description field here, as I don't think the descriptions here get run though the generators, I think they are meant to be static strings.

phenning replied:
Thinking about this some more I don't believe that the post action descriptions here ever actually get shown to a user by any of our existing hosts. I think this is more for informational purposes to the template author.  @vlada-shubina can confirm

phenning replied:
But to your point #2, I agree it should be about that item, but the item name is changeable at the time of creation and may not be called ValueController.cs when the user actually creates the file, so the original "created controller" is accurate from a different point of view.

In any event, this string will not be surfaced to the user.

dougbu replied:
> In any event, this string will not be surfaced to the user.

Ah, you heard back from @vlada-shubina 😄 In any case, I just tried to create a couple of items from our older templates that use the `primaryOutputs` name in the post-action `description` and confirmed this.

To be doubly sure, there isn't any logging somewhere that uses this resource (or a translation) in its output❔ I checked the output windows I have available in VS 17.6 Int Preview but, wasn't sure if there are options to turn something else on.

dougbu replied:
Reverted these changes done in next iteration&hellip;

vlada-shubina replied:
description may be shown to use in CLI context, if the post action fails. 
Therefore it is normally localised.

However, this post action doesn't run in CLI, so @phenning is correct here, but mind that it only applies when post action is not used in CLI.

sayedihashimi commented on src/ProjectTemplates/Web.ItemTemplates/content/RazorView/.template.config/template.json:
```diff
@@ -28,7 +28,7 @@
       "id": "openInEditor",
       "condition": "(HostIdentifier != \"dotnetcli\" && HostIdentifier != \"dotnetcli-preview\")",
-      "description": "Opens the created view in the editor",
+      "description": "Opens Index.cshtml in the editor",
```
This is also similar to `HomeController` and `ValueController`, the file that's generated will be determined by the user, `Index.cshtml` is not likely to make sense to the enduser.

dougbu committed:
Address code review comments
- no need to change post-action `description`s

dougbu committed:
!fixup! Missed one `ReadWrite` misspelling

dougbu commented:
/fyi The Arcade SDK adds a Microsoft.TemplateEngine.Authoring.Tasks package reference in all projects. That detects template.json files and propagate the strings into localize/templatestrings.*.json files.

dougbu commented on src/ProjectTemplates/Web.ItemTemplates/content/ApiController/.template.config/localize/templatestrings.en.json:
```diff
@@ -3,7 +3,9 @@
   "name": "API Controller",
   "description": "API Controller with or without read/write actions",
   "symbols/namespace/description": "namespace for the generated code",
+  "_symbols/namespace/description.comment": "Should this start w/ 'Namespace'?",
```
Reviewers, please answer questions like this in this file and the other two templatestrings.en.json files. Other `templatestrings.*.json` files can be ignored since they're just like the templatestrings.en.json files except for the comment entries.

dougbu commented on src/ProjectTemplates/Web.ItemTemplates/content/ApiController/.template.config/localize/templatestrings.en.json:
```diff
@@ -3,7 +3,9 @@
-  "postActions/openInEditor/description": "Opens ValueController.cs in the editor"
-}
+  "_symbols/actions/description.comment": "Should this start w/ 'Create'?",
+  "symbols/actions/displayName": "Add Read/Write Actions",
+  "postActions/openInEditor/description": "Opens the created controller in the editor"
```
To see exactly which strings I changes (as opposed to strings that were just newly copied from the template.json files automatically), look at changes in the template.json files. For example, I changed the `displayName` resource but not the post-action `description`.

mkArtakMSFT reviewed (approved):
Thanks @dougbu !

dougbu commented:
What are your thoughts on the questions I placed in the templatestrings.en.json files @mkArtakMSFT❔ (Not merging this w/o at least removing those comments.)

mkArtakMSFT commented on src/ProjectTemplates/Web.ItemTemplates/content/RazorView/.template.config/localize/templatestrings.en.json:
```diff
@@ -1,6 +1,7 @@
-  "postActions/openInEditor/description": "Opens Index.cshtml in the editor"
-}
+  "description": "An empty Razor view",
+  "_description.comment": "Reviewers: Could also be 'A Razor view without a page model'. And should View be capitalized?",
```
Let's capitalize the `View` but leave the text as is for now.

dougbu replied:
done

mkArtakMSFT commented on src/ProjectTemplates/Web.ItemTemplates/content/MvcController/.template.config/localize/templatestrings.en.json:
```diff
@@ -3,7 +3,9 @@
   "name": "MVC Controller",
   "description": "MVC Controller with or without read/write actions",
   "symbols/namespace/description": "namespace for the generated code",
+  "_symbols/namespace/description.comment": "Should this start w/ 'Namespace'?",
```
Yes, let's capitalize the first letters in all those sentences where you've raised the question.

dougbu replied:
done

mkArtakMSFT reviewed (approved):
Answered your questions

dougbu committed:
Capitalize as agreed in PR
- remove comment nodes from templatestrings.en.json files
- also, update English resources to match template.json content

dougbu committed:
Build to propagate strings
- fill in missed files in localize/ folder
